lib/galaxy/tools/parameters/basic.py

```diff
@@ -2038,7 +2038,7 @@ def append( list, id, hid, name, src, keep=False ):
         visible_hda = other_values.get( self.name )
         has_matched = False
         for hda in history.active_datasets_children_and_roles:
-            match = dataset_matcher.hda_match( hda )
+            match = dataset_matcher.hda_match( hda, check_implicit_conversions=workflow_building_modes.DISABLED, check_security=workflow_building_modes.DISABLED )
```

Comment:
This method takes booleans and you are sending it a named constant. I'd just change these to false... if not you could change the default values in the method signatures to be DISABLED - but these methods do not have to have anything to do with workflows so that seems off a little.